x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/search_index/pipelines/ml_inference/add_inference_pipeline_flyout.tsx

```diff
@@ -147,11 +149,29 @@ export const AddInferencePipelineHorizontalSteps: React.FC = () => {
     isPipelineDataValid,
   } = useValues(MLInferenceLogic);
   const { onAddInferencePipelineStepChange } = useActions(MLInferenceLogic);
+
+  /**
+   * Convenience function for determining the status of a step in the horizontal nav.
+   * @param currentStep The current step in the pipeline.
+   * @param otherStep The step to compare against.
+   * @returns The status of the step.
+   */
+  const getStepStatus = (
+    currentStep: AddInferencePipelineSteps,
+    otherStep: AddInferencePipelineSteps
+  ) => {
+    return currentStep > otherStep
+      ? 'complete'
```

Comment:
Is it a good idea to use constants instead of raw strings for status? For example,

const STATUS_COMPLETE = 'complete';
const STATUS_INCOMPLETE = 'incomplete';

Reply:
Good idea, although "complete" and "incomplete" are not arbitrary; they are valid `status` values for [EuiStepHorizontal](https://elastic.github.io/eui/#/navigation/steps#horizontal-steps) (check out the `Props` tab).

Normally we don't make constants for property values, however in this case the values come from a function that doesn't know about the valid options. Then again, this function is only used locally for this purpose, and so I don't see much value in adding a constant for more control.

I don't feel strongly about this, I can go either way.

Reply:
There is a type for this, although its not an enum, [EuiStepStatus](https://github.com/elastic/eui/blob/main/src/components/steps/step_number.tsx#L33-L43)

Reply:
I changed the function return type to `EuiStepStatus`.